src/biogeophys/BalanceCheckMod.F90

```diff
@@ -398,87 +572,178 @@ subroutine BalanceCheck( bounds, &
           end if
        end do
 
-       ! Water balance check
+       ! Water balance check at the column level
 
        do c = bounds%begc, bounds%endc
 
           ! add qflx_drain_perched and qflx_flood
           if (col%active(c)) then
 
-             errh2o(c) = endwb(c) - begwb(c) &
+             errh2o_col(c) = endwb_col(c) - begwb_col(c) &
                   - (forc_rain_col(c)        &
                   + forc_snow_col(c)         &
-                  + qflx_floodc(c)           &
-                  + qflx_sfc_irrig(c)        &
-                  + qflx_glcice_dyn_water_flux(c) &
-                  - qflx_evap_tot(c)         &
-                  - qflx_surf(c)             &
-                  - qflx_qrgwl(c)            &
-                  - qflx_drain(c)            &
-                  - qflx_drain_perched(c)    &
-                  - qflx_ice_runoff_snwcp(c) &
-                  - qflx_ice_runoff_xs(c)    &
-                  - qflx_snwcp_discarded_liq(c) &
-                  - qflx_snwcp_discarded_ice(c)) * dtime
+                  + qflx_flood_col(c)        &
+                  + qflx_sfc_irrig_col(c)    &
+                  + qflx_glcice_dyn_water_flux_col(c) &
+                  - qflx_evap_tot_col(c)     &
+                  - qflx_surf_col(c)         &
+                  - qflx_qrgwl_col(c)        &
+                  - qflx_drain_col(c)        &
+                  - qflx_drain_perched_col(c) &
+                  - qflx_ice_runoff_col(c)   &
+                  - qflx_snwcp_discarded_liq_col(c) &
+                  - qflx_snwcp_discarded_ice_col(c)) * dtime
 
           else
 
-             errh2o(c) = 0.0_r8
+             errh2o_col(c) = 0.0_r8
 
           end if
 
        end do
        
-       errh2o_max_val = maxval(abs(errh2o(bounds%begc:bounds%endc)))
+       errh2o_max_val = maxval(abs(errh2o_col(bounds%begc:bounds%endc)))
 
        if (errh2o_max_val > h2o_warning_thresh) then
 
-           indexc = maxloc( abs(errh2o(bounds%begc:bounds%endc)), 1 ) + bounds%begc -1
-           write(iulog,*)'WARNING:  water balance error ',&
+           indexc = maxloc( abs(errh2o_col(bounds%begc:bounds%endc)), 1 ) + bounds%begc - 1
+           write(iulog,*)'WARNING:  column-level water balance error ',&
              ' nstep= ',nstep, &
              ' local indexc= ',indexc,&
            ! ' global indexc= ',GetGlobalIndex(decomp_index=indexc, clmlevel=namec), &
-             ' errh2o= ',errh2o(indexc)
+             ' errh2o= ',errh2o_col(indexc)
          if ((errh2o_max_val > error_thresh) .and. (DAnstep > skip_steps)) then
               
-              write(iulog,*)'clm urban model is stopping - error is greater than 1e-5 (mm)'
+              write(iulog,*)'CTSM model stopping because errh2o > ', error_thresh, ' mm'
```

Comment:
- [ ] Thanks a lot for making this change. Dave (who labeled himself as the grammar police) pointed out that this should really just be 'CTSM is stopping', not 'CTSM model', since M in CTSM is already 'model'. Would you be willing to make that quick edit to this and other similar lines?

Reply:
Thanks for catching :-)